src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/services/RestrictedPatientCleanup.kt

```diff
@@ -14,7 +15,7 @@ class RestrictedPatientCleanup(
 
   @Transactional
   fun deleteRestrictedPatientOnExternalMovementIntoPrison(prisonerNumber: String) {
-    val restrictedPatient = restrictedPatientsRepository.findById(prisonerNumber).orElse(null) ?: return
+    val restrictedPatient = restrictedPatientsRepository.findByIdOrNull(prisonerNumber) ?: return
```

Comment:
It might be nice to log nulls here? In theory it shouldn't happen (like so many things in RP!) so it could help debugging if we know when it has happened.

Reply:
We are listening on **ALL** prison reception / offender movement events.  I would imagine most of the time they wouldn't concern restricted patients so the `return` would be the norm